File: Mobile-App/app/src/main/java/com/example/happyhabits/feature_application/feature_workout/data/model/WeightsForm.kt
package com.example.happyhabits.feature_application.feature_workout.data.model

import com.example.happyhabits.feature_application.feature_workout.domain.model.Exercise
import kotlinx.serialization.Serializable

@Serializable
data class WeightsForm(
    val userId: String,
    val date: String,
    val type: String = "Weights",
    val time: String,
    val duration: String,
    val notes: String? = "",
    val exercises: List<Exercise>
)

File: API-Server/Happy Habits App/Forms/WeightsForm.cs
using Happy_Habits_App.Model;
using MongoDB.Bson.Serialization.Attributes;
using System.Text.Json.Serialization;

namespace Happy_Habits_App.Forms
{
    public class WeightsForm : HabitForm
    {
        [JsonPropertyName("type")]
        public string? Type { get; set; } = null;
        [JsonPropertyName("time")]
        public string? Time { get; set; } = null;
        [JsonPropertyName("duration")]
        public string? Duration { get; set; } = null;
        [JsonPropertyName("notes")]
        public string? Notes { get; set; } = null;
        [JsonPropertyName("exercises")]
        public List<Exercise> Exercises { get; set; } = new List<Exercise>();

        public bool IsValid
        {
            get
            {
                return !string.IsNullOrEmpty(UserId) &&
                       !string.IsNullOrEmpty(Date) &&
                       !string.IsNullOrEmpty(Time) &&
                       !string.IsNullOrEmpty(Duration) &&
                       Exercises.Count > 0;
            }
        }
    }
}
